core/new-gui/src/app/dashboard/component/top-bar/user-icon/user-login/ngbdmodal-user-login.component.ts

```diff
@@ -99,6 +106,51 @@ export class NgbdModalUserLoginComponent implements OnInit {
       }, () => this.registerErrorMessage = 'Registration failed. Could due to duplicate username.');
   }
 
+  /**
+   * this method will init A Google Oauth instance and gapi
+   */
+  public async initGoogleOauth(): Promise<void> {
+    // load gapi
+    const gapiLoad = new Promise((resolve) => {
+      gapi.load('auth2', resolve);
+    });
+    // call gapi.auth2 init
+    return gapiLoad.then(async () => {
+      await gapi.auth2
+        .init({ client_id: environment.google.clientID })
+        .then(auth => {
+          this.oauthInstance = auth;
+          this.gapiSetUp = true;
+        });
+    });
+  }
+
+  /**
+   * this method will return the authorization code
+   */
+  public async authenticate(): Promise<void> {
+    // initialize gapi if not done yet
+    if (!this.gapiSetUp) {
+      await this.initGoogleOauth();
+    }
+    await this.oauthInstance.signIn().then(
+      user => this.user = user,
+      error => this.error = error
+    )
+    await this.oauthInstance.grantOfflineAccess().then(
```

Comment:
can you add some comments here to explain the `grantOfflineAccess`?

Reply:
done